src/components/FindMovie/FindMovie.tsx

```diff
@@ -1,10 +1,79 @@
-import React from 'react';
+import classNames from 'classnames';
+import React, { useState } from 'react';
+import { getMovie } from '../../api';
+import { Movie } from '../../types/Movie';
+import { MovieCard } from '../MovieCard';
 import './FindMovie.scss';
 
-export const FindMovie: React.FC = () => {
+type Props = {
+  onAdd: (newMovie: Movie) => void,
+};
+
+export const FindMovie: React.FC<Props> = ({ onAdd }) => {
+  const [query, setQuery] = useState('');
+  const [movie, setMovie] = useState<Movie | null>(null);
+  const [isError, setIsError] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const handleFormSubmit = async (event: React.FormEvent) => {
+    event.preventDefault();
+    setIsLoading(true);
+
+    try {
+      const normalizedQuery = query
+        .split(' ')
+        .map(word => word.trim())
+        .join(' ');
+      const response = await getMovie(normalizedQuery);
+
+      if ('Error' in response) {
+        setIsError(true);
+        throw new Error(
+          'There is no movie with such title, please, try one more time',
+        );
+      } else {
+        const {
+          Poster,
+          Title,
+          Plot,
+          imdbID,
+        } = response;
+        const poster = Poster === 'N/A'
+          ? 'https://via.placeholder.com/360x270.png?text=no%20preview'
+          : Poster;
+        const newMovie = {
+          title: Title,
+          description: Plot,
+          imgUrl: poster,
+          imdbUrl: `https://www.imdb.com/title/${imdbID}`,
+          imdbId: imdbID,
+        };
+
+        setIsLoading(false);
+        setMovie(newMovie);
+      }
+    } catch (error) {
+      setIsError(true);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  const hadleAddButton = () => {
+    if (movie !== null) {
+      onAdd(movie);
+    }
```

Comment:
Add button is visible on the page only when a movie isn't null, so this check is redundant.